{
    "id": "dbpedia_4791_2",
    "rank": 22,
    "data": {
        "url": "https://nmap.org/changelog.html",
        "read_more_link": "",
        "language": "en",
        "title": "Nmap Change Log",
        "top_image": "",
        "meta_img": "",
        "images": [
            "https://nmap.org/shared/images/nst-icons.svg#menu",
            "https://nmap.org/shared/images/nst-icons.svg#close",
            "https://nmap.org/images/sitelogo.png, /images/sitelogo-2x.png 2x",
            "https://nmap.org/shared/images/nst-icons.svg#search",
            "https://nmap.org/shared/images/nst-icons.svg#search",
            "https://nmap.org/shared/images/nst-icons.svg#twitter",
            "https://nmap.org/shared/images/nst-icons.svg#facebook",
            "https://nmap.org/shared/images/nst-icons.svg#github",
            "https://nmap.org/shared/images/nst-icons.svg#reddit"
        ],
        "movies": [],
        "keywords": [],
        "meta_keywords": [
            "Nmap",
            "changelog",
            "nmap changes",
            "github issues",
            "nmap bugs",
            "nmap update",
            "nmap version"
        ],
        "tags": null,
        "authors": [],
        "publish_date": null,
        "summary": "",
        "meta_description": "List of the most recent changes to the free Nmap Security Scanner",
        "meta_lang": "en",
        "meta_favicon": "/shared/images/tiny-eyeicon.png",
        "meta_site_name": "",
        "canonical_link": null,
        "text": "[Zenmap]Fixed a bug in the Compare Scans window of Zenmap on OS X resulting in this message instead of Ndiff output:\n\nImportError: dlopen(/Applications/Zenmap.app/Contents/Resources/lib/python2.7/lib-dynload/datetime.so, 2): no suitable image found. Did find: /Applications/Zenmap.app/Contents/Resources/lib/python2.7/lib-dynload/datetime.so: mach-o, but wrong architecture\n\nReported by Kyle Gustafson. [Daniel Miller]\n\n[Zenmap]Avoid file corruption in zenmap.conf, reported as files containing many null (\"\\x00\") characters. Example exceptions:\n\nTypeError: int() argument must be a string or a number, not 'list' ValueError: unable to parse colour specification\n\n[Zenmap]Fix a crash due to incorrect file paths when installing to /usr/local prefix. Example:\n\nException: File '/home/blah/.zenmap/scan_profile.usp' does not exist or could not be found!\n\n[Zenmap][GH#235]Fix several failures to launch Zenmap on OS X. The new build process eliminates these errors:\n\nIOError: [Errno 2] No such file or directory: '/Applications/Zenmap.app/Contents/Resources/etc/pango/pangorc.in' LSOpenURLsWithRole() failed for the application /Applications/Zenmap.app with error -10810.\n\n[Zenmap]Fixed the Zenmap .dmg installer for OS X. Zenmap failed to launch, producing this error:\n\nCould not import the zenmapGUI.App module: 'dlopen(/Applications/Zenmap.app/Contents/Resources/lib/python2.6/lib-dynload/glib/_glib.so, 2): Library not loaded: /Users/david/macports-10.5/lib/libffi.5.dylib\\n Referenced from: /Applications/Zenmap.app/Contents/Resources/lib/python2.6/lib-dynload/glib/_glib.so\\n Reason: image not found'.\n\n[Zenmap]Fixed a bug which caused this crash message:\n\nIOError: [Errno socket error] [Errno 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond\n\nThe bug was caused by us adding a DOCTYPE definition to Nmap's XML output which caused Python's XML parser to try and fetch the DTD every time it parses an XML file. We now override that DTD-fetching behavior. [Daniel Miller]\n\n[Zenmap]Removed a crashing error that could happen when canceling a \"Print to File\" on Windows:\n\nTraceback (most recent call last): File \"zenmapGUI\\MainWindow.pyo\", line 831, in _print_cb File \"zenmapGUI\\Print.pyo\", line 156, in run_print_operation GError: Error from StartDoc\n\nThis bug was reported by Imre AdÃ¡csi. [David Fifield]\n\nFixed a bug that caused an incorrect source address to be set when scanning certain addresses (apparently those ending in .0) on Windows XP. The symptom of this bug was the messages\n\nget_srcaddr: can't connect socket: The requested address is not valid in its context. Failed to convert source address to presentation format!?! Error: Unknown error\n\nThanks to Robert Washam and Jorge Hernandez for reports and help debugging. [David Fifield]\n\nFixed an error that occurred when scanning certain addresses like 192.168.0.0 on Windows XP:\n\nget_srcaddr: can't connect socket: The requested address is not valid in its context. nexthost: failed to determine route to 10.80.0.0\n\n[David Fifield]\n\n[Nmap, Ncat, Nping]The default unit for time specifications is now seconds, not milliseconds, and times may have a decimal point. 1000 now means 1000 seconds, or about 17 minutes, not 1000 milliseconds. Floating point values such as 1.5 are now allowed. This affects the following options: Nmap:\n\n--host-timeout --max-rtt-timeout --min-rtt-timeout --initial-rtt-timeout --scan-delay --max-scan-delay --stats-every\n\nNcat:\n\n-d --delay -i --idle-timeout -w --wait\n\nNping:\n\n--delay --host-timeout --icmp-orig-time --icmp-recv-time --icmp-trans-time\n\nSome sanity checks have been added to catch what looks like an attempt to use the old millisecond defaults. For example, --host-timeout 10000 yields\n\nSince April 2010, the default unit for --host-timeout is seconds, so your time of \"10000\" is 2.8 hours. If this is what you want, use \"10000s\". QUITTING!\n\nYou can always disable the warning by giving an explicit unit.\n\nNmap now honors routing table entries that override interface addresses and netmasks. For example, with this configuration:\n\n************************INTERFACES************************ DEV (SHORT) IP/MASK TYPE UP MAC eth0 (eth0) 192.168.0.21/24 ethernet up 00:00:00:00:00:00 . **************************ROUTES************************** DST/MASK DEV GATEWAY 192.168.0.3/32 eth0 192.168.0.1 192.168.0.0/24 eth0\n\nNmap will not consider 192.168.0.3 directly connected through eth0, even though it matches the interface's netmask. It won't try to ARP ping 192.168.0.3, but will route traffic through 192.168.0.1.\n\nARP requests now work with libpcap Linux \"cooked\" encapsulation. According to http://wiki.wireshark.org/SLL, this encapsulation is used on devices \"where the native link layer header isn't available or can't be used.\" Before this, attempting any ARP operation on such an interface would fail with the error\n\nread_arp_reply_pcap called on interfaces that is datatype 113 rather than DLT_EN10MB (1)\n\n[David]\n\n[NSE]Fixed http-enum so it uses the full path name for the fingerprints file. This prevents it from quitting with an error like this:\n\nNSE: http-enum: Attempting to parse fingerprint file nselib/data/http-fingerprints NSE: http-enum against 10.99.24.140:443 threw an error! C:\\Program Files\\Nmap\\scripts\\http-enum.nse:198: bad argument #1 to 'lines' (nselib/data/http-fingerprints: No such file or directory) stack traceback:\n\n[Kris, Brandon, Ron Meldau]\n\n[NSE]Fixed a bug in http.lua that could lead to an assertion failure. It happened when there was an error getting the a response at the beginning of a batch in http.pipeline. The symptoms of the bug were:\n\nNSE: Received only 0 of 1 expected reponses. Decreasing max pipelined requests to 0. NSOCK (0.1870s) Write request for 0 bytes... nmap: nsock_core.c:516: handle_write_result: Assertion `bytesleft > 0' failed.\n\nThe error was reported by Brandon Enright and pyllyukko.\n\nFixed a bug in traceroute that could lead to a crash:\n\nterminate called after throwing an instance of 'std::out_of_range' what(): bitset::test\n\nIt happened when the preliminary distance guess for a target was greater than 30, the size of an internal data structure. David and Brandon tracked down the problem.\n\n[Zenmap]Fixed a crash that could occur when entering certain characters in the target entry (those whose UTF-8 encoding contains a byte that counts as whitespace in the Windows locale):\n\nFile \"zenmapGUI\\ScanNotebook.pyo\", line 184, in _target_entry_changed File \"zenmapCore\\NmapOptions.pyo\", line 719, in render_string UnicodeDecodeError: 'utf8' codec can't decode byte 0xc3 in position 1: unexpected end of data\n\nFor more details on this curious problem, see http://seclists.org/nmap-dev/2009/q4/82 [David]\n\n[Nsock]Made it a fatal error instead of silent memory corruption when an attempt is made to use a file descriptor whose number is not less than FD_SETSIZE. This applies only on non-Windows platforms where FD_SETSIZE is a limit on the value of file descriptors as well as a limit on the number of descriptors in the set. The error will look like\n\nnsock_core.c:186: Attempt to FD_SET fd 1024, which is not less than FD_SETSIZE (1024). Try using a lower parallelism.\n\nThanks to Brandon Enright for discovering the problem and much help debugging it, and to Jay Fink for submitting an initial patch. [David]\n\nNmap now handles the case when a primary network interface (venet0) does not have an address assigned but its aliases do (venet0:1 etc.). This could result in the error messages\n\nFailed to find device venet0 which was referenced in /proc/net/route Failed to lookup subnet/netmask for device (venet0): venet0: no IPv4 address assigned\n\nThis was observed under OpenVZ. [Dmitry Levin]\n\n[Zenmap]Tweaked the file format of the topology icons (firewall.png, padlock.png, etc.) in an attempt to improve compatibility with some versions of GTK+. This may fix a crash like\n\nFile \"radialnet/gui/Image.py\", line 53, in get_pixbuf self.__cache[icon + image_type] = gtk.gdk.pixbuf_new_from_file(file) GError: Couldn't recognize the image file format for file 'radialnet/padlock.png'\n\nThanks to Trevor Bain for a report and help debugging. [David]\n\nNsock handles a certain Windows connect error, WSAEADDRNOTAVAIL (errno 10049), preventing an assertion failure that looked like\n\nStrange connect error from 203.65.42.255 (10049): No such file or directory Assertion failed: 0, file .\\src\\nsock_core.c, line 290\n\nThe error could be seen by running a version scan against a broadcast address. Thanks to Tilo KÃ¶ppe and James Liu for reporting the problem. [David]\n\n[Zenmap]Several corner-case crashes were fixed: [David]\n\nFile \"radialnet\\gui\\NodeNotebook.pyo\", line 429, in __create_widgets KeyError: 'tcp' File \"radialnet\\gui\\RadialNet.pyo\", line 1531, in __livens_up AttributeError: 'NoneType' object has no attribute 'get_nodes' File \"zenmapGUI\\MainWindow.pyo\", line 308, in _create_ui_manager GError: Odd character '\\' File \"radialnet/gui/ControlWidget.py\", line 104, in __create_widgets AttributeError: 'module' object has no attribute 'STOCK_INFO' File \"radialnet\\util\\integration.pyo\", line 385, in make_graph_from_hosts KeyError: 'hops'\n\nNmap now provides progress statistics in the XML output in verbose mode. Here are some examples of the format (etc is \"estimated time until completion) and times are in UNIX time_t (seconds since 1970) format.\n\n<taskbegin task=\"SYN Stealth Scan\" time=\"1151384685\" /> <taskprogress task=\"SYN Stealth Scan\" time=\"1151384715\" percent=\"13.85\" remaining=\"187\" etc=\"1151384902\" /> <taskend task=\"SYN Stealth Scan\" time=\"1151384776\" /> <taskbegin task=\"Service scan\" time=\"1151384776\" /> <taskend task=\"Service scan\" time=\"1151384788\" />\n\nThanks to Adam Vartanian (flooey(a)gmail.com) for the patch.\n\nWhen debugging (-d) is specified, Nmap now prints a report on the timing variables in use. Thanks to Doug Hoyte for the patch. The report loos like this:\n\n---------- Timing report ---------- hostgroups: min 1, max 100000 rtt-timeouts: init 250, min 50, max 300 scan-delay: TCP 5, UDP 1000 parallelism: min 0, max 0 max-retries: 2, host-timeout 900000 -----------------------------------\n\nNmap now prints the number (and total bytes) of raw IP packets sent and received when it completes, if verbose mode (-v) is enabled. The report looks like:\n\nNmap finished: 256 IP addresses (3 hosts up) scanned in 30.632 seconds Raw packets sent: 7727 (303KB) | Rcvd: 6944 (304KB)\n\nImplemented a service classification scheme to separate the vendor/product name from the version number and any extra info that is provided. Instead of v/[big version string]/, the new match lines include v/[vendor/productname]/[version]/[extrainfo]/ . See the docs at the top of nmap-service-probes for more info. This doesn't change the normal output (which lumps them together anyway), but they are separate in the XML so that higher-level programs can easily match against just a product name. Here are a few examples of the improved service element:\n\n<service name=\"ssh\" product=\"OpenSSH\" version=\"3.1p1\" extrainfo=\"protocol 1.99\" method=\"probed\" conf=\"10\" /> <service name=\"domain\" product=\"ISC Bind\" version=\"9.2.1\" method=\"probed\" conf=\"10\" /> <state state=\"open\" /><service name=\"rpcbind\" version=\"2\" extrainfo=\"rpc #100000\" method=\"probed\" conf=\"10\" /> <service name=\"rndc\" method=\"table\" conf=\"3\" />\n\nUpdated XML output to handle new version and service detection information. Here are a few examples of the new output:\n\n<port protocol=\"tcp\" portid=\"22\"><state state=\"open\" /><service name=\"ssh\" version=\"OpenSSH 3.1p1 (protocol 1.99)\" method=\"probed\" conf=\"10\" /></port> <port protocol=\"tcp\" portid=\"111\"><state state=\"open\" /><service name=\"rpcbind\" version=\"2 (rpc #100000)\" method=\"probed\" conf=\"10\" /></port> <port protocol=\"tcp\" portid=\"953\"><state state=\"open\" /><service name=\"rndc\" method=\"table\" conf=\"3\" /></port>\n\nApplied patch by Martin Kluge (martin(a)elxsi.info) which adds --ttl option, which sets the outgoing IPv4 TTL field in packets sent via all raw scan types (including ping scans and OS detection). The patch \"should work\" on Windows, but hasn't been tested. A TTL of 0 is supported, and even tends to work on a LAN:\n\n14:17:19.474293 192.168.0.42.60214 > 192.168.0.40.135: S 326:326(0) [ttl 0] 14:17:19.474456 192.168.0.40.135 > 192.168.0.42.60214: S 280:280(0) ack 326 (ttl 128)"
    }
}